### PR TITLE
Add Account Api Keys use case

### PR DESCRIPTION
**Pull request by dolcalmi**
- [X] Add account api key schema
- [X] Add api key creation
- [X] Add get api keys by account
- [X] Disable api key by label
- [x] Add Graphql mutations/queries
- [x] Add middleware and update context
- [ ] ~~Update current mutations/queries to use authenticated account~~ Will be done once we update walletPublicId

### CODE REVIEW

File: src/app/accounts/add-api-key-for-account.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { hashApiKey, randomApiKey } from "@domain/accounts"
+import { AccountApiKeysRepository } from "@services/mongoose"
+
+export const addApiKeyForAccount = async ({
+  accountId,
+  label = "",
```
Comment (bodymindarts) [inline src/app/accounts/add-api-key-for-account.ts:6]: IMO defaults shouldn't be set at the use case level. Client should decide what the default should be.

File: src/app/accounts/index.types.d.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+type AddApiKeyForAccountArgs = {
+  accountId: AccountId
+  label?: string
```
Comment (bodymindarts) [inline src/app/accounts/index.types.d.ts:3]: No optional arguments for use-cases please. Optional arguments on the use-case level are a smell that point to there actually being more than 1 use case in play.

File: src/domain/accounts/api-keys.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+const hasExpired = (expireAt: Date): boolean => {
+  const now = new Date(Date.now())
+  return expireAt < now
+}
+
+export const randomApiKey = async (
+  expireAt: Date,
+  label?: string,
```
Comment (bodymindarts) [inline src/domain/accounts/api-keys.ts:26]: Shouldn't be optional - defaults should be set higher up.

File: test/integration/services/mongoose/account-api-keys.spec.ts [hunk 4]
```diff
@@ -0,0 +1,106 @@
+import { hashApiKey, randomApiKey } from "@domain/accounts"
+import { CouldNotFindError } from "@domain/errors"
+import { AccountApiKeysRepository } from "@services/mongoose"
+
+const repo = AccountApiKeysRepository()
+const delta = 30 // days
```
Comment (bodymindarts) [inline test/integration/services/mongoose/account-api-keys.spec.ts:6]: how about making the comment part of the variable name: `expiration_in_days` or `delta_days` or something like hat.

**Review by bodymindarts — COMMENTED** [review]
It lgtm but I'm not sure about the key/secret both being returned. Most API's that I know of just have 1 value as the api secret. I don't really see the use case for having 2.

File: test/integration/services/mongoose/account-api-keys.spec.ts (hunk 4 above)
Comment (dolcalmi) [reply test/integration/services/mongoose/account-api-keys.spec.ts]: fixed

File: src/domain/accounts/api-keys.ts (hunk 3 above)
Comment (dolcalmi) [reply src/domain/accounts/api-keys.ts:26]: fixed

**Review by samerbuna — DISMISSED** [review]

**Review by vindard — DISMISSED** [review]
Looks good, needs rebase

**Review by samerbuna — DISMISSED** [review]

**Review by samerbuna — APPROVED** [review]

**Review by vindard — APPROVED** [review]

**Pull request merged**